From c3a9db8bf5fc853f4ffba9d1bc1f88902f9ca523 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:50 -0400
Subject: [ethernet] i40e: fix passing wrong error code to i40e_open()

Message-id: <1409658042-22689-110-git-send-email-sassmann@redhat.com>
Patchwork-id: 90302
O-Subject: [RHEL7.1 PATCH 109/281] i40e: fix passing wrong error code to i40e_open()
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jean Sacren <sakiwit@gmail.com>

The commit 6c167f582ea9 ("i40e: Refactor and cleanup i40e_open(),
adding i40e_vsi_open()") introduced a new function i40e_vsi_open()
with the regression by a typo. Due to the commit, the wrong error
code would be passed to i40e_open(). Fix this error in
i40e_vsi_open() by turning the macro into a negative value so that
i40e_open() could return the pertinent error code correctly.

Fixes: 6c167f582ea9 ("i40e: Refactor and cleanup i40e_open(), adding i40e_vsi_open()")
Signed-off-by: Jean Sacren <sakiwit@gmail.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit ce9ccb17ef5b5088172f46dd246c92523fd3a524)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 1819f26..78f4f07 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4333,7 +4333,7 @@ int i40e_vsi_open(struct i40e_vsi *vsi)
     dev_driver_string(&pf->pdev->dev));
   err = i40e_vsi_request_irq(vsi, int_name);
  } else {
-  err = EINVAL;
+  err = -EINVAL;
   goto err_setup_rx;
  }
 
-- 
1.7.1